tests/TestCase/Controller/AudioControllerTest.php

```diff
@@ -46,4 +60,28 @@ public function accessesProvider() {
     public function testAudioControllerAccess($url, $user, $response) {
         $this->assertAccessUrlAs($url, $user, $response);
     }
+
+    public function testAudioDownload_ok() {
+        $this->initAudioStorageDir();
+
+        $audioFileContents = $this->createAudioFile(1);
+        $this->get('/en/audio/download/1');
+        $this->assertResponseOk();
+        $this->assertResponseEquals($audioFileContents);
+        $this->assertHeader('Content-Disposition', 'attachment; filename="3-1.mp3"');
+
+        $this->initAudioStorageDir();
```

Comment:
Well, I don’t want `initAudioStorageDir()` to be called before/after each and every unit test of this class, because it would be a waste of time and resource. It is only needed in tests that manipulate audio files.

Reply:
I'm just wondering about the implications of doing "init, do stuff, init again" vs. "init, do stuff, restore state before init".

Reply:
Thanks for clarifying. That code indeed leaves an empty folder.